JavaScript/JavaScriptSDK/Telemetry/PageViewManager.ts

```diff
@@ -0,0 +1,118 @@
+                name = window.document && window.document.title || "";
+            }
+
+            if (typeof url !== "string") {
+                url = window.location && window.location.href || "";
+            }
+
+            var pageViewSent = false;
+            var customDuration = 0;
+
+            if (Telemetry.PageViewPerformance.isPerformanceTimingSupported()) {
+                var start = Telemetry.PageViewPerformance.getPerformanceTiming().navigationStart;
+                customDuration = Telemetry.PageViewPerformance.getDuration(start, +new Date);
+            } else {
+                this.appInsights.sendPageViewInternal(
+                    name,
+                    url,
+                    0,
+                    properties,
+                    measurements);
+                this.appInsights.flush();
+                pageViewSent = true;
+            }                     
+
+            if (this.overridePageViewDuration || !isNaN(duration)) {
+                // 1, 2, 4 cases
+                this.appInsights.sendPageViewInternal(
+                    name,
+                    url,
+                    duration ? duration : customDuration,
+                    properties,
+                    measurements);
+                this.appInsights.flush();
+                pageViewSent = true;
+            }
+
+            var maxDurationLimit = 60000;
+
+            if (!Telemetry.PageViewPerformance.isPerformanceTimingSupported()) {
+                // no navigation timing (IE 8, iOS Safari 8.4, Opera Mini 8 - see http://caniuse.com/#feat=nav-timing)
+                _InternalLogging.throwInternalNonUserActionable(LoggingSeverity.CRITICAL,
+                    "trackPageView: navigation timing API used for calculation of page duration is not supported in this browser. This page view will be collected without duration and timing info.");
+                return;
+            }
+
+            var handle = setInterval(() => {
+                try {
+                    if (Telemetry.PageViewPerformance.isPerformanceTimingDataReady()) {
+                        clearInterval(handle);
+                        var pageViewPerformance = new Telemetry.PageViewPerformance(name, url, null, properties, measurements);
+
+                        if (!pageViewPerformance.getIsValid() && !pageViewSent) {
+                            // If navigation timing gives invalid numbers, then go back to "override page view duration" mode.
+                            // That's the best value we can get that makes sense.
+                            this.appInsights.sendPageViewInternal(name, url, customDuration, properties, measurements);
+                            this.appInsights.flush();
+                        } else {
+                            if (!pageViewSent) {
+                                this.appInsights.sendPageViewInternal(name, url, pageViewPerformance.getDurationMs(), properties, measurements);
```

Comment:
I could, but its unnecessary - timeout is alreday cleared, and noone down below tries to send it (without entering the timeout again)